Changes 2026

nmarticorena opened the pull request:
This PR incorporates the changes for 2026 including the transition to pixi and adding the new capture script.

Some of the pendings: 
- [x] Update PenguinPi-robot once https://github.com/qcr/PenguinPi-robot/pull/6 is merged
- [x] Add windows to platforms
- [x] Update markdowns to include pixi install

TedVanderfeen committed:
Pynput doesn't work through WSL. Using a matplotlib window to capture keyboard inputs instead

nmarticorena committed:
Add pixi lock and versions for osx m

nmarticorena committed:
Merge branch 'pynput'

nmarticorena committed:
Add fix of matplotlib option when using osx backend

nmarticorena committed:
Add wsl and osx related ignores

nmarticorena committed:
add linux-64 as a platform

nmarticorena committed:
test matplotlib collect in linux and add better prints

nmarticorena committed:
Add cuda version and test_install

nmarticorena committed:
Add win-64

nmarticorena committed:
update PenguinPi-robot

nmarticorena committed:
Update instructions

nmarticorena committed:
Add bypass for windows, and reorganize the instructions

nmarticorena committed:
Update instructions, pixi install -a fails if there is not cuda available in linux or windows

Tobias-Fischer commented on instructions/Preparation.md:
```diff
@@ -1,45 +1,75 @@
+Pixi's biggest strength is its ability to create reproducible, powerful, and flexible workspaces. A workspace lives in a directory on your system, and is a collection of Pixi environments that can be used to develop one or many projects in that directory. The dependencies and enviroments are defined in a [manifest file](../pixi.toml).
 
-Nothing that can be done with conda cannot be achieved otherwise. However, with conda it is usually easier and cross-platform.
+In order to install the dependencies you need to run the following command in the cloned RVSS_Need4Speed directory.
```
```suggestion
In order to install the dependencies you need to run the following command in the cloned `RVSS_Need4Speed` directory.
```

Tobias-Fischer commented on instructions/Preparation.md:
```diff
@@ -1,45 +1,75 @@
+pixi install # Will install default environment in your computer
 
-Conda is also an environment manager (like `virtualenv`). Therefore, if an environment is ruined beyond repair, you can just remove it and start over with a clean one.
+pixi install -e cuda # Install gpu dependencies
```
Please add a comment that this will only work for CUDA machines, on both Linux and Windows.

Tobias-Fischer commented on instructions/Preparation.md:
```diff
@@ -1,45 +1,75 @@
-- We strongly recommend using the community-driven `conda-forge` channel, instead of the `defaults` channel that is maintained by Anaconda. 
-- The easiest way to install conda with conda-forge as default channel is with `Miniforge3`. You can download the installers from [here](https://github.com/conda-forge/miniforge#miniforge3).
-- If you are on Linux/MacOS, simply type `sh Miniforge3-*.sh` and follow the instructions. On Windows, double click `Miniforge3-Windows-x86_64.exe` and follow the instructions.
+*In Windows you may need to enable [long path lengths](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=powershell#application-manifest-updates-to-declare-long-path-capability) if you get an error when creating the environment.*
```
Also add a comment that user names with spaces, Chinese characters etc can cause trouble, and to try a short path like `C:\dev` or sth like that

Tobias-Fischer commented on instructions/Preparation.md:
```diff
@@ -1,45 +1,75 @@
+
+
+# Download the starter data
+Navigate to the [huggingface dataset](https://huggingface.co/datasets/dimitym/RVSS_Need4Speed_starter/tree/main) and download data.zip. Place this inside your cloned repository and extract the zip file.
```
Please create a pixi task that does this automatically using wget or something like that (just add a dependency on it).

Tobias-Fischer commented on instructions/Preparation.md:
```diff
@@ -1,45 +1,75 @@
+# Download the starter data
+Navigate to the [huggingface dataset](https://huggingface.co/datasets/dimitym/RVSS_Need4Speed_starter/tree/main) and download data.zip. Place this inside your cloned repository and extract the zip file.
+
+You should have a `data` folder that contains a `train_starter` and `val_starter` folder, containing 793 and 436 images respectively.
```
Remove this comment once we have a pixi task

Tobias-Fischer commented on instructions/Preparation.md:
```diff
@@ -1,45 +1,75 @@
+
+*In windows you may need to run the command as `powershell -ExecutionPolicy Bypass -c "pixi shell" ` if running scripts are disabled in the system*
+
+## Check instalation
```
```suggestion
## Check installation
```

Tobias-Fischer commented on instructions/Preparation.md:
```diff
@@ -1,45 +1,75 @@
+
+## Check instalation
+
+After the instalation is done you can check your install by running `pixi run test` or alternatively `pixi run -e cuda test` to check if the gpu is supported.
```
```suggestion
After the installation is done, you can check your install by running `pixi run test` or alternatively `pixi run -e cuda test` to check if the gpu is supported.
```

Tobias-Fischer commented on pixi.toml:
```diff
@@ -0,0 +1,54 @@
+pytorch-gpu = ">=2.9.1,<3"
+
+
+[feature.test.tasks]
```
Is this really correct? Why do we need a `test` feature? We can simply use `[tasks]` can't we?

nmarticorena replied:
This was in favor of https://pixi.prefix.dev/latest/tutorials/multi_environment/#using-multiple-environments-to-test-multiple-versions-of-a-package to promt the user to choose and env but it does only works if cpu is not the default env, so I will remove it

Tobias-Fischer commented on pixi.toml:
```diff
@@ -0,0 +1,54 @@
+
+
+[feature.test.tasks]
+test = {cmd = "python test_install.py"}
```
Can you please expand the test script to check if the dataset download is successful?

Tobias-Fischer reviewed (changes requested):
Hi @nmarticorena - thanks a lot for putting this together! Some minor comments :)

nmarticorena committed:
Add get_dataset task

nmarticorena committed:
fix typo

nmarticorena committed:
Fallback to old collect script, it does not work in windows, tested natively and not in wsl

nmarticorena committed:
Avoid overwritting README.md and .gitattributes

nmarticorena committed:
Add check dataset

nmarticorena committed:
Address comments

Tobias-Fischer commented on pixi.toml:
```diff
@@ -0,0 +1,58 @@
+[workspace]
+authors = [""]
```
Please add the folks that have previously worked on this here (Dim, you, me, ..)

Tobias-Fischer commented on pixi.toml:
```diff
@@ -0,0 +1,58 @@
+[tasks]
+test = {cmd = "python test_install.py"}
+download_data = {cmd = "hf download dimitym/RVSS_Need4Speed_starter data.zip --repo-type dataset --local-dir ."}
+get_dataset = {cmd = "7za x ./data.zip", depends-on = ["download_data"]}
```
Can we add some more tasks here? Test motors, train model, ... ?

nmarticorena replied:
I did not add them so students could use Pixi as an env. I can add all the Python scripts as tasks with the respective env variables, but I am not sure if this will confuse the students more. 

Perhaps only a task to test the motors? and we let the students to run python directly for the rest of files?

Tobias-Fischer replied:
Yes sure. Maybe in this case we should add a reference to setting up pixi in VS code / selecting the right environment?

Tobias-Fischer reviewed (changes requested):
Nearly there!

nmarticorena committed:
Add Authors

nmarticorena committed:
fix typo

nmarticorena committed:
Update password of starlink

nmarticorena commented on instructions/ControlRobot.md:
```diff
@@ -2,7 +2,8 @@
 
-Make sure you are connected to RVSS_Starlink, with password RVSS2025 before you attempt to control the robot.
+
+Make sure you are connected to RVSS_Starlink, with password RVSS2026 before you attempt to control the robot.
```
@Tobias-Fischer Here I can change the instruction in favour of using the hotspots of the robots, that will leave us with a fixed ip that we can add to the default tasks. 

I will just need to update how to get the ssid of your group robot so let me know what will you prefer

nmarticorena replied:
I added some comments in the robot connection readme

nmarticorena reviewed (commented):
Some comments to discuss

nmarticorena committed:
add interpreter instructions

nmarticorena committed:
Add conection handling commands

nmarticorena committed:
Add extra instructions regarding robot connection

Tobias-Fischer committed:
Revise ControlRobot.md for clarity and formatting

Updated formatting and improved clarity in instructions for connecting and controlling the robot.

Tobias-Fischer committed:
Revise pixi installation and usage instructions

Updated instructions for installing and using pixi, including clarifications on commands and environment setup.
